### PR TITLE
Fix non-ASCII file names on older R Windows

### PR DESCRIPTION
**Pull request by gaborcsardi**
`normalizePath()` on R 3.4 and before converts the
path to the native encoding, apparently. (Nothing in
the docs about this.)

The C code of brio assumes that paths are in UTF-8,
so we need to explicitly convert them, for that to
hold on older R versions.

Tested from R 3.3 to R 4.1.

### CODE REVIEW

File: tests/testthat/test-utf8.R [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+test_that("UTF-8 file names", {
+  skip_on_cran()
```
Comment (jimhester) [inline tests/testthat/test-utf8.R:3]: Why did you need to skip this on CRAN?

Comment (gaborcsardi) [reply tests/testthat/test-utf8.R:3]: No specific reason, but they are updating a lot of things wrt encodings nowadays, and a couple of weird things have popped up recently.

Comment (gaborcsardi) [reply tests/testthat/test-utf8.R:3]: OTOH it would make sense to test brio on R-devel Windows, on the CI.

Comment (jimhester) [reply tests/testthat/test-utf8.R:3]: I would prefer not to skip it unless we have a specific reason.

Sure we can test on R-devel for windows as well.

Comment (gaborcsardi) [reply tests/testthat/test-utf8.R:3]: Yeah, it is up to you, really. If we don't skip, then make sure you test on Solaris and a non-UTF-8 Linux as well before submitting to CRAN, just in case. These platforms might not work with non-ASCII characters at all.

Or we can make it Windows-only.
